src/mpi/init/mpir_init.c

```diff
@@ -433,6 +436,8 @@ int MPII_Finalize(MPIR_Session * session_ptr)
     mpi_errno = MPIR_Async_things_finalize();
     MPIR_ERR_CHECK(mpi_errno);
 
+    MPIR_Pset_free();
+
```

Comment:
Either move this changes into the earlier commit where `MPIR_Pset_init/finalize` is defined or split into a new commit.

Reply:
Split into new commit.